source/common/http/header_map_impl.h

```diff
@@ -230,6 +232,8 @@ class HeaderMapImpl : public virtual HeaderMap, NonCopyable {
   HeaderList headers_;
   // This holds the internal byte size of the HeaderMap.
   uint64_t cached_byte_size_ = 0;
+
+  friend HeaderListView;
```

Comment:
This shouldn't need to be a friend class or be defined in terms of the impl at all. You should be able to generate the view using the public API entirely.

Reply:
All we need is to extract all of keys on HeaderMap for [this](https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/common/expr/context.h#L87). I think that if we prohibit accessing HeaderList directly (or via view like class) we can't achieve this objective... Is there a critical approach to access HeaderList without changing implementation?

Reply:
HeaderMap::iterate should work.

Reply:
Exactly.

Reply:
done